pitop/pma/servo_motor.py

```diff
@@ -46,16 +50,16 @@ def zero_point(self):
 
     @zero_point.setter
     def zero_point(self, zero_position):
-        hardware_min_angle = -ServoHardwareSpecs.ANGLE_RANGE / 2
-        hardware_max_angle = ServoHardwareSpecs.ANGLE_RANGE / 2
-        if not (hardware_min_angle <= zero_position <= hardware_max_angle):
-            raise ValueError(f"Value must be from {hardware_min_angle} to {hardware_max_angle} degrees (inclusive)")
+        if not (self._HARDWARE_MIN_ANGLE <= zero_position <= self._HARDWARE_MAX_ANGLE):
+            raise ValueError(f"Value must be from {self._HARDWARE_MIN_ANGLE} to {self._HARDWARE_MAX_ANGLE} degrees "
+                             f"(inclusive)")
 
         self._zero_point = zero_position
-        self.min_angle = hardware_min_angle - self._zero_point
-        self.max_angle = hardware_max_angle - self._zero_point
+        self._min_angle = self._min_angle - self._zero_point
+        self._max_angle = self._max_angle - self._zero_point
```

Comment:
I think we'll lose reference of `_min_angle` and `_max_angle` in here by using the current angles to calculate the new ones...

thinking about it, maybe there are 2 approaches for zero-point setting:
- referencing the "hardware" zero, meaning, the argument `zero_position` refers to the hardware angle, where for example, 0 will always be 0 and it won't change over time
- referencing the latest known zero point setting (and valid angle range), which might differ from the hardware zero (and the original hardware min-max angles). In this approach, the `zero_position` argument might be outside of the hardware angle range...

As this function is now, I think we're mixing these to approaches, since we check against hardware angles at first, but the use the current min-max angles to calculate the new range... I think we should decide on the approach, and document it, since it should be clear to the user!

Reply:
Good spot, that would've been a serious bug!